boms/user/client/ejb-client/pom.xml

```diff
@@ -74,6 +80,14 @@
                             <bomWithDependencies>true</bomWithDependencies>
                             <licenses>true</licenses>
                             <inheritExclusions>UNMANAGED</inheritExclusions>
+                            <channels>
+                                <channel>
+                                    <manifest>
+                                        <groupId>org.wildfly.channels</groupId>
+                                        <artifactId>wildfly-ee</artifactId>
+                                    </manifest>
```

Comment:
Guys, this is a problem, specifically the versionless manifest coordinates.

You can see the problem even in the CI run for this PR https://github.com/wildfly/wildfly/actions/runs/11478610181/job/31943175993#step:5:20008 :
```
...
[INFO] Building WildFly BOM Builders: Client EJB Builder 35.0.0.Beta1-SNAPSHOT [129/194]
...
[INFO] --- wildfly-bom-builder-plugin:2.0.7.Final:build-bom (build-bom) @ wildfly-ejb-client-bom-builder ---
...
Oct 23, 2024 11:13:38 AM org.wildfly.channel.maven.VersionResolverFactory$MavenResolverImpl resolveChannelMetadata
INFO: Resolving channel metadata from Maven artifact org.wildfly.channels:wildfly-ee:34.0.0.Final
```

I would expect `org.wildfly.channels:wildfly-ee:35.0.0.Beta1-SNAPSHOT` is resolved here, otherwise the generated BOMs are completely off wrt current build.

Reply:
reported at https://issues.redhat.com/browse/WFLY-19908

Reply:
BTW even if the resolution is somehow fixed to look at local repository too, we might still need to explicitly set version here otherwise the build won't be guaranteed to be reproducible anymore - eg. when building 34.0.x for whatever reason, the 35.0.0.Beta1 might be already public and as higher version it will be resolved here which is clearly wrong.